app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :tweets
+  has_many :images
+  has_many :comments
+  has_and_belongs_to_many :followers, join_table: 'followers_followings', class_name: 'User', foreign_key: 'followee_id', association_foreign_key: 'follower_id'
```

Comment:
Let's comment habtm and do this by has many through also

Reply:
Also, tell the difference between the two and when to use which

Reply:
done

Reply:
When we have a join table with some attributes other than reference columns then we use `has_many through`
otherwise we use `has_and_belongs_to_many`